modules/data.land/tests/testthat/test-extract_soil_nc.R

```diff
@@ -90,16 +89,147 @@ test_that("extract_soil_gssurgo handles ensemble generation", {
     lat = 40.1164,
     lon = -88.2434,
     size = 3,
-    grid_size = 3,
-    grid_spacing = 100,
-    depths = c(0.15, 0.30)
+    radius = 500,
+    depths = c(0, 0.15, 0.30)
   )
   
   expect_false(is.null(res))
   
   expect_type(res, "list")
-  expect_equal(length(res), 4)
+  expect_equal(length(res), 5)
+  file_paths <- unlist(res)
+  expect_true(all(file.exists(file_paths)))
+})
+
+test_that("extract_soil_gssurgo works with custom AOI polygon", {
+  skip_on_cran()
+  skip_on_ci()
+  skip_if_not_installed("terra")
+  skip_if_not_installed("sf")
+  
+  tmp_outdir <- withr::local_tempdir("gssurgo_test_")
+  
+  # Create small polygon AOI
+  aoi_coords <- matrix(c(
+    -88.25, 40.11,
+    -88.24, 40.11,
+    -88.24, 40.12,
+    -88.25, 40.12,
+    -88.25, 40.11
+  ), ncol = 2, byrow = TRUE)
+  
+  aoi <- terra::vect(aoi_coords, type = "polygons", crs = "epsg:4326")
+  
+  res <- extract_soil_gssurgo(
+    outdir = tmp_outdir,
+    aoi = aoi,
+    size = 2,
+    depths = c(0, 0.15, 0.30)
+  )
+  
+  expect_false(is.null(res))
+  expect_type(res, "list")
+  expect_gt(length(res), 0)
   
   file_paths <- unlist(res)
   expect_true(all(file.exists(file_paths)))
+})
+
+test_that("extract_soil_gssurgo handles different buffer radii", {
+  skip_on_cran()
+  skip_on_ci()
+  tmp_outdir <- withr::local_tempdir("gssurgo_test_")
+  
+  # Small radius
+  res_small <- extract_soil_gssurgo(
+    outdir = tmp_outdir,
+    lat = 40.1164,
+    lon = -88.2434,
+    size = 1,
+    radius = 200,
+    depths = c(0, 0.15)
+  )
+  
+  # Larger radius (should potentially capture more mukeys)
+  res_large <- extract_soil_gssurgo(
+    outdir = tmp_outdir,
+    lat = 40.1164,
+    lon = -88.2434,
+    size = 1,
+    radius = 1000,
+    depths = c(0, 0.15)
+  )
+  
+  expect_false(is.null(res_small))
+  expect_false(is.null(res_large))
+  expect_type(res_small, "list")
+  expect_type(res_large, "list")
```

Comment:
checking both for null and for list type looks redundant to me -- wouldn't a null result be caught by expect_type as not a list?